spec/models/user_spec.rb

```diff
@@ -138,23 +138,32 @@ def user_error_message(*keys)
     end
 
     describe "#user_fields" do
-      fab!(:user_field) { Fabricate(:user_field) }
+      fab!(:user_field) { Fabricate(:user_field, show_on_profile: true) }
       fab!(:watched_word) { Fabricate(:watched_word, word: "bad") }
 
       before { user.set_user_field(user_field.id, value) }
 
       context "when user fields contain watched words" do
         let(:value) { "bad user field value" }
 
-        it "is not valid" do
-          user.valid?
-          expect(user.errors[:base].size).to eq(1)
-          expect(user.errors.messages[:base]).to include(/you can't post the word/)
+        context "when user field is public" do
+          it "is not valid" do
+            user.valid?
+            expect(user.errors[:base].size).to eq(1)
+            expect(user.errors.messages[:base]).to include(/you can't post the word/)
+          end
+        end
+
+        context "when user field is private" do
+          before { user_field.update(show_on_profile: false) }
+
+          it { is_expected.to be_valid }
         end
       end
 
-      context "when user fields do not contain watched words" do
-        let(:value) { "good user field value" }
+      context "when private user fields contain watched words" do
```

Comment:
We can remove this block now (leftover) :sweat_smile:

Reply:
🤦‍♂️